### PR TITLE
Converted samples to use dialog.open and dialog.submit [Don't Merge]

### PR DESCRIPTION
**Pull request by Prithvi-MSFT**
Converted samples to use dialog.open and dialog.submit

### CODE REVIEW

**Review by Pawan-MSFT — APPROVED** [review]
looks correct, Approving!

**Comment by Harikrishnan-MSFT** [comment]
Build Pipeline - 
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/10299000795
image

File: samples/bot-task-module/nodejs/README.md [hunk 1]
```diff
@@ -13,24 +13,24 @@ extensions:
 urlFragment: officedev-microsoft-teams-samples-bot-task-module-nodejs
 ---
 
-# Teams Task Module
+# Teams Dialog
```
Comment (Wajeed-msft) [inline samples/bot-task-module/nodejs/README.md:16]:
```suggestion
# Teams Dialog (referred as task modules in TeamsJS v1.x)
```

File: samples/bot-task-module/csharp/README.md [hunk 2]
```diff
@@ -13,24 +13,24 @@ extensions:
 urlFragment: officedev-microsoft-teams-samples-bot-task-module-csharp
 
 ---
-# Teams Task Module
+# Teams Dialog
```
Comment (Wajeed-msft) [inline samples/bot-task-module/csharp/README.md:16]:
```suggestion
# Teams Dialogs (referred as task modules in TeamsJS v1.x)
```

**Review by Wajeed-msft — APPROVED** [review]

File: samples/bot-task-module/nodejs/README.md (hunk 1 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-task-module/nodejs/README.md]: Updated

File: samples/bot-task-module/csharp/README.md (hunk 2 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-task-module/csharp/README.md]: Updated

File: README.md [hunk 3]
```diff
@@ -216,7 +216,7 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
 
 |    | Sample Name    | Description                                                                      | C#    | JavaScript      |App Manifest|
 |:--:|:---------------|:---------------------------------------------------------------------------------|:------|:----------------|:-------|
-|1|Task Modules          | Sample app showing off the Teams Task Module, a way to invoke custom code from a bot, a tab, or both! |[View][bot-task-module#cs]     |[View][bot-task-module#js] ![toolkit-icon](assets/toolkit-icon.png) |[View](/samples/bot-task-module/csharp/demo-manifest/bot-task-module.zip)
+|1|Teams Dialog         | Sample app showing off the Teams Dialog, a way to invoke custom code from a bot, a tab, or both! |[View][bot-task-module#cs]     |[View][bot-task-module#js] ![toolkit-icon](assets/toolkit-icon.png) |[View](/samples/bot-task-module/csharp/demo-manifest/bot-task-module.zip)
```
Comment (Wajeed-msft) [inline README.md:219]:
```suggestion
|1|Teams Dialogs (referred as task modules in TeamsJS v1.x)         | Sample app showing off the Teams Dialog, a way to invoke custom code from a bot, a tab, or both! |[View][bot-task-module#cs]     |[View][bot-task-module#js] ![toolkit-icon](assets/toolkit-icon.png) |[View](/samples/bot-task-module/csharp/demo-manifest/bot-task-module.zip)
```

File: samples/bot-task-module/csharp/Pages/HelloWorld.cshtml [hunk 4]
```diff
@@ -5,4 +5,4 @@
 }
 <p>
     Welcome to Microsoft Teams Task Module Sample App (C#)
```
Comment (Wajeed-msft) [inline samples/bot-task-module/csharp/Pages/HelloWorld.cshtml:7]:
```suggestion
    Welcome to Microsoft Teams Dialogs (referred as task modules in TeamsJS v1.x) Sample App (C#)
```

File: samples/bot-task-module/csharp/README.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 ---
 page_type: sample
-description: This sample app demonstrate how to use Task Module using Bot Framework v4
+description: This sample app demonstrate how to use Dialog using Bot Framework v4
```
Comment (Wajeed-msft) [inline samples/bot-task-module/csharp/README.md:3]:
```suggestion
description: This sample app demonstrate how to use Dialogs (referred as task modules in TeamsJS v1.x) using Bot Framework v4
```

File: samples/bot-task-module/csharp/README.md [hunk 6]
```diff
@@ -13,24 +13,24 @@ extensions:
 urlFragment: officedev-microsoft-teams-samples-bot-task-module-csharp
 
 ---
-# Teams Task Module
+# Teams Dialog (referred as task modules in TeamsJS v1.x)
 
-Bot Framework Teams Task Module sample.
+Bot Framework Teams Dialog sample.
```
Comment (Wajeed-msft) [inline samples/bot-task-module/csharp/README.md:18]:
```suggestion
Bot Framework Teams Dialogs (referred as task modules in TeamsJS v1.x) sample.
```

File: samples/bot-task-module/csharp/README.md [hunk 7]
```diff
@@ -13,24 +13,24 @@ extensions:
...
 
-This bot has been created using [Bot Framework](https://dev.botframework.com). It shows how to fetch a Task Module from Hero Card or Adaptive Card buttons and receive input from the Task Module in the bot.
+This bot has been created using [Bot Framework](https://dev.botframework.com). It shows how to fetch a Dialog from Hero Card or Adaptive Card buttons and receive input from the Dialog in the bot.
 
 ## Included Features
 * Bots
 * Tabs
-* Task Modules
+* Dialogs
```
Comment (Wajeed-msft) [inline samples/bot-task-module/csharp/README.md:25]:
```suggestion
* Dialogs (referred as task modules in TeamsJS v1.x)
```

File: samples/bot-task-module/csharp/README.md [hunk 8]
```diff
@@ -13,24 +13,24 @@ extensions:
...
 
 ## Interaction with bot
 ![bot-task-module ](Images/Bot_Tab_TaskModule.gif)
 
 ## Try it yourself - experience the App in your Microsoft Teams client
 Please find below demo manifest which is deployed on Microsoft Azure and you can try it yourself by uploading the app manifest (.zip file link below) to your teams and/or as a personal app. (Sideloading must be enabled for your tenant, [see steps here](https://docs.microsoft.com/microsoftteams/platform/concepts/build-and-test/prepare-your-o365-tenant#enable-custom-teams-apps-and-turn-on-custom-app-uploading)).
 
-**Teams Task Module:** [Manifest](/samples/bot-task-module/csharp/demo-manifest/bot-task-module.zip)
+**Teams Dialog:** [Manifest](/samples/bot-task-module/csharp/demo-manifest/bot-task-module.zip)
```
Comment (Wajeed-msft) [inline samples/bot-task-module/csharp/README.md:33]:
```suggestion
**Teams Dialogs (referred as task modules in TeamsJS v1.x):** [Manifest](/samples/bot-task-module/csharp/demo-manifest/bot-task-module.zip)
```

File: samples/bot-task-module/csharp/assets/sample.json [hunk 9]
```diff
@@ -2,11 +2,11 @@
   {
     "name": "officedev-microsoft-teams-samples-bot-task-module-csharp",
     "source": "officeDev",
-    "title": "Task Module using Bot Framework v4",
-    "shortDescription": "This sample app demonstrate how to use Task Module using Bot Framework v4",
+    "title": "Dialog using Bot Framework v4",
```
Comment (Wajeed-msft) [inline samples/bot-task-module/csharp/assets/sample.json:5]:
```suggestion
    "title": "Dialogs (referred as task modules in TeamsJS v1.x) using Bot Framework v4",
```

File: samples/bot-task-module/nodejs/README.md [hunk 10]
```diff
@@ -1,6 +1,6 @@
 ---
 page_type: sample
-description: This is an sample application which showcases how to invoke task module from adaptive and hero cards.
+description: This is an sample application which showcases how to invoke Dialog from adaptive and hero cards.
```
Comment (Wajeed-msft) [inline samples/bot-task-module/nodejs/README.md:3]:
```suggestion
description: This is an sample application which showcases how to invoke Dialogs (referred as task modules in TeamsJS v1.x) from adaptive and hero cards.
```

File: samples/bot-task-module/nodejs/assets/sample.json [hunk 11]
```diff
@@ -2,11 +2,11 @@
   {
     "name": "officedev-microsoft-teams-samples-bot-task-module-nodejs",
     "source": "officeDev",
-    "title": "Bot Task Module",
-    "shortDescription": "This is an sample application which showcases how to invoke task module from adaptive and hero cards.",
+    "title": "Bot Dialog",
```
Comment (Wajeed-msft) [inline samples/bot-task-module/nodejs/assets/sample.json:5]:
```suggestion
    "title": "Bot Dialogs (referred as task modules in TeamsJS v1.x)",
```

File: README.md (hunk 3 above)
Comment (Harikrishnan-MSFT) [reply README.md:219]: Updated

File: samples/bot-task-module/csharp/Pages/HelloWorld.cshtml (hunk 4 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-task-module/csharp/Pages/HelloWorld.cshtml:7]: Updated

File: samples/bot-task-module/csharp/README.md (hunk 5 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-task-module/csharp/README.md:3]: Updated

File: samples/bot-task-module/csharp/README.md (hunk 6 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-task-module/csharp/README.md:18]: Updated

File: samples/bot-task-module/csharp/README.md (hunk 7 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-task-module/csharp/README.md:25]: Updated

File: samples/bot-task-module/csharp/README.md (hunk 8 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-task-module/csharp/README.md:33]: Updated

File: samples/bot-task-module/csharp/assets/sample.json (hunk 9 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-task-module/csharp/assets/sample.json:5]: Updated

File: samples/bot-task-module/nodejs/README.md (hunk 10 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-task-module/nodejs/README.md:3]: Updated

File: samples/bot-task-module/nodejs/assets/sample.json (hunk 11 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-task-module/nodejs/assets/sample.json:5]: Updated

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**